examples/jira_mysql_install.pp

```diff
@@ -0,0 +1,24 @@
+node default {
+
+  class { '::mysql::server':
+    root_password    => 'strongpassword',
+  } ->
+
+  mysql::db { 'jira':
+    user     => 'jiraadm',
+    password => 'mypassword',
+    host     => 'localhost',
+    grant    => ['ALL'],
+  } ->
+
+  class { '::jira':
+    javahome    => '/opt/java/latest',
+    db          => 'mysql',
+    dbport      => '3306',
+    dbdriver    => 'com.mysql.jdbc.Driver',
+    dbtype      => 'mysql',
+  } ->
+
+  class { '::jira::facts': }
```

Comment:
I deeply resent the antipattern of declaring classes with the resource like syntax when no parameters are being passed. Does the `jira::facts` class really depend on the `jira` class? Otherwise I'd prefer to drop the `->` and use `include`.

Generally, is there a style guide that addresses this question? The [official docs](https://docs.puppetlabs.com/puppet/latest/reference/lang_classes.html#include-like-vs-resource-like) have a discussion at least, but no clear recommendation I think.